lib/install/validate/validator.go

```diff
@@ -161,18 +156,27 @@ func CreateNoDCCheck(ctx context.Context, input *data.Data) (*Validator, error)
 		return nil, errors.New(detail)
 	}
 
-	return v, nil
+	return v, v.datacenter(allowEmptyDC)
 }
 
-func (v *Validator) datacenter() error {
-	if v.Session.Datacenter == nil {
-		detail := "Datacenter must be specified in --target (e.g. https://addr/datacenter)"
-		log.Error(detail)
-		v.suggestDatacenter()
-		return errors.New(detail)
+func (v *Validator) datacenter(allowEmpty bool) error {
+	if allowEmpty && v.DatacenterPath == "" {
+		return nil
+	}
+	if v.Session.Datacenter != nil {
+		v.DatacenterPath = v.Session.Datacenter.InventoryPath
+		return nil
+	}
+	var detail string
+	if v.DatacenterPath != "" {
+		detail = fmt.Sprintf("Datacenter %q in --target is not found", strings.TrimPrefix(v.DatacenterPath, "/"))
+	} else {
+		// this means multiple datacenter exists, but user did not specify it in --target
+		detail = "Datacenter must be specified in --target (e.g. https://addr/datacenter)"
 	}
-	v.DatacenterPath = v.Session.Datacenter.InventoryPath
-	return nil
+	log.Error(detail)
```

Comment:
I understand your point, but current suggestion methods logging out message. We should have a separate PR for that refactor.